### PR TITLE
Add deny list transactional tests

### PR DESCRIPTION
**Pull request by lxfind**
## Description 

This PR replaces the deny list simtest with transactional tests, which are much easier to inspect and validate.
To make errors easier to visualize and update, this PR also introduces a new function in the test adapter that can replace known addresses and IDs with names/fake ids in the error string.

## Test Plan 

CI

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-transactional-test-runner/src/test_adapter.rs [hunk 1]
```diff
@@ -991,6 +992,28 @@ impl<'a> MoveTestAdapter<'a> for SuiTestAdapter<'a> {
...
+        for (name, account) in &self.accounts {
+            let addr = account.address.to_string();
+            let replace = format!("[{}]", name);
+            err = err.replace(&addr, &replace);
+            // Also match without 0x since different error messages may use different format.
+            err = err.replace(&addr[2..], &replace);
+        }
+        for (id, fake_id) in &self.object_enumeration {
+            let id = id.to_string();
+            let replace = format!("[object({})]", fake_id);
```
Comment (tnowacki) [inline crates/sui-transactional-test-runner/src/test_adapter.rs:1010]:
```suggestion
            let replace = format!("object({})", fake_id);
```

File: crates/sui-transactional-test-runner/src/test_adapter.rs [hunk 2]
```diff
@@ -991,6 +992,28 @@ impl<'a> MoveTestAdapter<'a> for SuiTestAdapter<'a> {
...
     }
+
+    /// Process the error string such that it's less dependent on specific addresses or object IDs. Instead, they are
+    /// replaced by the account names or fake IDs as much as possible. This reduces the effort of updating tests
+    /// when something changed.
+    async fn process_error(&self, error: anyhow::Error) -> anyhow::Error {
+        let mut err = error.to_string();
+        for (name, account) in &self.accounts {
+            let addr = account.address.to_string();
+            let replace = format!("[{}]", name);
```
Comment (tnowacki) [inline crates/sui-transactional-test-runner/src/test_adapter.rs:1003]:
```suggestion
            let replace = format!("@{}", name);
```

File: crates/sui-adapter-transactional-tests/tests/entry_points/obj_vector.exp [hunk 3]
```diff
@@ -33,7 +33,7 @@ mutated: object(0,0), object(5,0)
 gas summary: computation_cost: 1000000, storage_cost: 6011600,  storage_rebate: 2249676, non_refundable_storage_fee: 22724
 
 task 7 'run'. lines 132-136:
-Error: Error checking transaction input objects: InvalidChildObjectArgument { child_id: 0x2a3551a605e9b22e289b35067b1aaa934c563a6bc2de5cc049658ce8ce89b3c7, parent_id: 0x692d988f96ea464e0f265006c68ec4a81831402dde7a75cae73197e07768a461 }
+Error: Error checking transaction input objects: InvalidChildObjectArgument { child_id: [object(6,0)], parent_id: [object(6,1)] }
```
Comment (tnowacki) [inline crates/sui-adapter-transactional-tests/tests/entry_points/obj_vector.exp:36]: These are signing errors I take it? We should really be passing back a string result here, and not relying on the anyhow error

Comment (lxfind) [reply crates/sui-adapter-transactional-tests/tests/entry_points/obj_vector.exp:36]: Yes.
lol you don't like the brackets?

Comment (lxfind) [reply crates/sui-adapter-transactional-tests/tests/entry_points/obj_vector.exp:36]: Would you prefer me to revert the error processing part or leave it as it is?

Comment (tnowacki) [reply crates/sui-adapter-transactional-tests/tests/entry_points/obj_vector.exp:36]: Leaving it as is fine, mostly just noting that it isn't really expected that you should have to do this

**Review by tnowacki — APPROVED** [review]
I don't like the brackets ;) But otherwise things look good! 

Should/do we have a test for sponsored transactions?

**Pull request merged**